shell/common/rasterizer.cc

```diff
@@ -355,7 +356,15 @@ void Rasterizer::FireNextFrameCallbackIfPresent() {
   callback();
 }
 
-void Rasterizer::SetResourceCacheMaxBytes(int max_bytes) {
+void Rasterizer::SetResourceCacheMaxBytes(size_t max_bytes, bool from_user) {
```

Comment:
This API is timing sensitive and depends on the surface already being acquired. It would be better if the cache size in bytes was instead stored in an `std::optional<size_t> on the rasterizer.`. Then, here and in `Rasterizer::Setup`, you could use `value_or(some_default)` to set the cache size of context of the surface. Similarly, the getter would then just the cache size from the optional ivar instead of depending on the surface being present. This scheme would also make the setting survive rasterizer teardown and re-setup.

Reply:
done